src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets

```diff
@@ -145,13 +145,27 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <UsingTask TaskName="Microsoft.NET.Build.Tasks.ResolvePackageDependencies"
              AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <UsingTask TaskName="Microsoft.NET.Build.Tasks.CheckForTargetInAssetsFile"
+           AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
 
   <!-- The condition on this target causes it to be skipped during design-time builds if
         the restore operation hasn't run yet.  This is to avoid displaying an error in
         the Visual Studio error list when a project is created before NuGet restore has
         run and created the assets file. -->
   <Target Name="RunResolvePackageDependencies"
           Condition=" '$(DesignTimeBuild)' != 'true' Or Exists('$(ProjectAssetsFile)')">
+    
+    <!-- Verify that the assets file has a target for the right framework.  Otherwise, if we restored for the
+         wrong framework, we'd end up finding no references to pass to the compiler, and we'd get a ton of
+         compile errors.
+         
+         The RuntimeIdentifier isn't used to select references to pass to the compiler, so we don't include
+         it here. -->
+    <CheckForTargetInAssetsFile
+      AssetsFilePath="$(ProjectAssetsFile)"
+      TargetFrameworkMoniker="$(TargetFrameworkMoniker)"
+      RuntimeIdentifier="" />
```

Comment:
Originally I was passing the `RuntimeIdentifier` parameter here.  However, that caused the `It_passes_ridless_target_to_compiler` test to fail.  I didn't see an obvious way to keep the test coverage except to not check the RID before the compiler is called.

The `GenerateBuildRuntimeConfigurationFiles` will check that the right RID has been restored.

I'm not entirely sure whether we should just check everything up-front and remove the `It_passes_ridless_target_to_compiler` test, or whether there's a better way to get the same coverage.